### ISSUE TITLE
tailscaled binary size increase in 1.56

### ISSUE THREAD

**Issue by kszaq** [tailscale/tailscale#10627]
### What are you trying to do?

From release to release, Tailscale is growing in features but also in executable size. This is not an issue for devices with lots of space but can be on memory-constrained devices, such as home routers.

I am building [smaller binaries](https://tailscale.com/kb/1207/small-tailscale) for arm64 router with:
```
GOOS=linux GOARCH=arm64 ./build_dist.sh --extra-small --box tailscale.com/cmd/tailscaled
```
Result:
```
-rwxr-xr-x  1 kszaq kszaq 19791872 tailscaled.combined.arm64.v1.54.1
-rwxr-xr-x  1 kszaq kszaq 23592960 tailscaled.combined.arm64.v.1.56.1 +19.2%
```
Static binaries from https://pkgs.tailscale.com/stable/#static :
```
-rwxr-xr-x 1 kszaq kszaq 16048093 Nov 30 19:46 tailscale_1.54.1_arm64/tailscale
-rwxr-xr-x 1 kszaq kszaq 29911769 Nov 30 19:46 tailscale_1.54.1_arm64/tailscaled
-rwxr-xr-x 1 kszaq kszaq 15992745 Dec 15 20:47 tailscale_1.56.1_arm64/tailscale -0.4%
-rwxr-xr-x 1 kszaq kszaq 42710851 Dec 15 20:47 tailscale_1.56.1_arm64/tailscaled +42.8%
```
Static binaries stripped:
```
-rwxr-xr-x 1 kszaq kszaq 10986216 Dec 16 11:40 tailscale_1.54.1_arm64/tailscale
-rwxr-xr-x 1 kszaq kszaq 20417304 Dec 16 11:40 tailscale_1.54.1_arm64/tailscaled
-rwxr-xr-x 1 kszaq kszaq 10921512 Dec 16 11:40 tailscale_1.56.1_arm64/tailscale -0.6%
-rwxr-xr-x 1 kszaq kszaq 29468536 Dec 16 11:40 tailscale_1.56.1_arm64/tailscaled +44.3%
```

### How should we solve this?

Provide more flags to exclude features in addition to currently supported:
`ts_omit_aws,ts_omit_bird,ts_omit_tap,ts_omit_kube`

**Comment by knyar** [comment]
I believe we are tracking client binary size improvements in #1278. If you have specific ideas around what dependencies can become optional and what impact that would have on binary size, could you please leave a comment there?

Barring further comment, we'll expect to close this during the next regular bug scrub.

**Comment by kszaq** [comment]
I was firstly looking why 1.56 is much bigger than 1.54. A quick bisect shows that it grew with e25f114916b4d8ced60ae2fd77a5982a826f2e63:
```
-rwxr-xr-x 1 kszaq kszaq 19791872 Dec 18 18:53 tailscaled.2f01d5e3dafb47b5b5e990cab6c0fd54b5c5d821
-rwxr-xr-x 1 kszaq kszaq 23592960 Dec 18 18:51 tailscaled.e25f114916b4d8ced60ae2fd77a5982a826f2e63
```
I'm very new to golang world, so I can only suspect it's due to a new dependency pulled into `tailscaled` static binary?

**Comment by andrew-d** [comment]
(cc @awly who authored #10507)

**Comment by knyar** [comment]
Thank you for identifying the reason for the increase, @kszaq! I've repurposed this issue to focus specifically on the increase in 1.56. If you have any broader thoughts/ideas on decreasing binary size, let's use #1278 for them.

**Comment by awly** [comment]
This should be resolved in 1.58 (likely mid-January).
While problematic, I don't think it's critical enough to make a 1.56.2 patch release.
Please let me know if this is causing real impact and we can build an unstable release as a temporary workaround!

**Comment by kszaq** [comment]
Thank you for a reapply quick fix! 

For me personally no bugfix release now is not an issue, since I'm building myself. 🙂
